Mathlib/Order/Filter/Basic.lean

```diff
@@ -1279,6 +1279,11 @@ theorem eventually_inf_principal {f : Filter α} {p : α → Prop} {s : Set α}
   mem_inf_principal
 #align filter.eventually_inf_principal Filter.eventually_inf_principal
 
+theorem eventually_iff_all_subsets {f : Filter α} {p : α → Prop} :
+    (∀ᶠ x in f, p x) ↔ ∀ (s : Set α), ∀ᶠ x in f, x ∈ s → p x :=
+  ⟨(fun h _ ↦ by filter_upwards [h] with _ pa _ using pa),
+  (fun h ↦ by filter_upwards [h univ] with _ pa using (pa (by simp)))⟩
```

Comment:
This is quite hard to read with all the brackets, so I'd recommend:
```suggestion
    (∀ᶠ x in f, p x) ↔ ∀ s : Set α, ∀ᶠ x in f, x ∈ s → p x where
  mp h _ := by filter_upwards [h] with _ pa _ using pa
  mpr h := by filter_upwards [h univ] with _ pa using (pa (by simp))
```

Reply:
Oh nice, I did not know we could do that!

Reply:
Why does it works? Doesn't this notation refer to typeclasses?